sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Generated/ContainerRegistryBlobRestClient.cs

```diff
@@ -35,23 +35,6 @@ public ContainerRegistryBlobRestClient(ClientDiagnostics clientDiagnostics, Http
             _url = url ?? throw new ArgumentNullException(nameof(url));
         }
 
-        internal HttpMessage CreateGetBlobRequest(string name, string digest)
```

Comment:
Is there a way to approach the change to generated methods that doesn't require re-implementing them manually?

Reply:
That would be great, but this is typically the tradeoff when needing to change the shape of generated requests. Do you have any suggestions to accomplish this in a better way?

Reply:
Could there be a way to indicate to the generator that there's a need to follow redirects in an API?

Reply:
We sometimes need to work with the  generator team to get them to use new features we add to Core.  It might be worth initiating that conversation.  Is this something that might needed in a fully generated client?

So as not to block this on that functionality, I'd also be open to a customization we could make in the outer HCL client if we could find a way to do it using that approach

Reply:
I think this is a bit of an edge case. For clients that want this configured globally (see below), we have an option now. This only pertains to clients which need this on a per-operation basis. ACR is the only example, I believe.

https://github.com/Azure/azure-sdk-for-net/blob/6ad5dbbb1b5240e1aa7d9bb5dfd912371d69ddac/sdk/communication/Azure.Communication.CallAutomation/src/CustomBuildHttpPieline.cs#L18-L19

Reply:
Ah, interesting about the global option.  Why did we decide not to enable it globally on ACR?

Reply:
We certainly could do that if you feel better about that. I was just trying to minimize the need for redirects. I'll defer to your judgement here.

Reply:
Help me understand this better, though, if you don't mind, since I'm still not sure what the best path forward is.  If the goal is to minimize the need for redirects, this implementation addresses that by only enabling redirects on messages for operations that can turn a 3xx response (is that correct?  I'm assuming that's how you chose the operations to customize, but I could totally be wrong).  But if all other operations don't return 3xx (or would throw if a non-2xx response is returned), is there a problem with enabling it for them too?  The latter is a serious question -- not rhetorical -- if there's a problem, I would want to know it before saying I think it's the right thing to enable it globally.  What risks do we take on if we were to do that?

Reply:
> Help me understand this better, though, if you don't mind, since I'm still not sure what the best path forward is. If the goal is to minimize the need for redirects, this implementation addresses that by only enabling redirects on messages for operations that can turn a 3xx response (is that correct? I'm assuming that's how you chose the operations to customize, but I could totally be wrong). 

Correct

> But if all other operations don't return 3xx (or would throw if a non-2xx response is returned), is there a problem with enabling it for them too? The latter is a serious question -- not rhetorical -- if there's a problem, I would want to know it before saying I think it's the right thing to enable it globally. What risks do we take on if we were to do that?

In theory, if we know there will never be a redirect, we don't care if redirects are enabled. However, the core issue is that for untrusted URIs, we can be fail safe that an unexpected redirect will not be followed by default, hence the desire to minimize exposure to this.

Reply:
> In theory, if we know there will never be a redirect, we don't care if redirects are enabled. However, the core issue is that for untrusted URIs, we can be fail safe that an unexpected redirect will not be followed by default, hence the desire to minimize exposure to this.

Yeah, I think it would be safe to assume that even if the swagger says the service won't return a 3xx, we could maybe get one.  But, it sounds like we feel safe that if we did get a 3xx for some operation where it's not allowed, the generated code would have us throw before it was followed, so we'd be safe there too?

Reply:
Is there a way we could intercept that?  Are there pipeline configurations we could make that might help us achieve the goal here?